services/app-api/src/handlers/third_party_API_authorizer.ts

```diff
@@ -0,0 +1,68 @@
+import type {
+    APIGatewayAuthorizerResult,
+    APIGatewayTokenAuthorizerEvent,
+    PolicyDocument,
+    APIGatewayTokenAuthorizerHandler,
+} from 'aws-lambda'
+import { newJWTLib } from '../jwt'
+
+// Hard coding this for now, next job is to run this config to this app.
+const jwtLib = newJWTLib({
+    issuer: 'fakeIssuer',
+    signingKey: 'notrandom',
+    expirationDurationS: 90 * 24 * 60 * 60, // 90 days
+})
+
+export const main: APIGatewayTokenAuthorizerHandler = async (
+    event
+): Promise<APIGatewayAuthorizerResult> => {
+    const authToken = event.authorizationToken.replace('Bearer ', '')
+    try {
+        // authentication step for validating JWT token
+        const userId = await jwtLib.userIDFromToken(authToken)
+
+        if (userId instanceof Error) {
+            const msg = 'Invalid auth token'
+            console.error(msg)
+
+            return generatePolicy('', event)
+        }
+
+        console.info({
+            message: 'third_party_API_authorizer succeeded',
+            operation: 'third_party_API_authorizer',
+            status: 'SUCCESS',
+        })
+
+        return generatePolicy(userId, event)
+    } catch (err) {
+        console.error('Invalid auth token. err => ', err)
```

Comment:
Can you change this error message to be more like "unexpected exception" or something? an invalid token should just be handed with the error on line 24, I think if we are catching here it's because something very weird happened.